### ISSUE TITLE
[ICE] no type for expr at Box<Any>

### ISSUE THREAD

**Issue by Grimy** [rust-lang/rust#25962]
I encountered an internal compiler error while trying out the official binary release of Rust 1.0.0. I was able to reduce it to the following code :

    0.a[0+0]

How to reproduce
-----------------------

    echo 'fn main(){0.a[0+0]}' | env RUST_BACKTRACE=1 rustc -

Expected result
--------------------

The compiler reports the invalid code without crashing.

Actual result
----------------

```
src/main.rs:1:11: 1:14 error: attempted access of field `a` on type `_`, but no field with that name was found
src/main.rs:1 fn main(){0.a[0+0]}
                        ^~~
error: internal compiler error: no type for expr in fcx 0x7f63e03eef88
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at 'Box<Any>', /home/rustbuild/src/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libsyntax/diagnostic.rs:209

stack backtrace:
   1:     0x7f63e788e449 - sys::backtrace::write::hbc46dc0cfb3b9537d4r
   2:     0x7f63e7896156 - panicking::on_panic::h74d3c14d86c58ac8jrw
   3:     0x7f63e7859462 - rt::unwind::begin_unwind_inner::h382cea404b11eb00t6v
   4:     0x7f63e4e313fd - rt::unwind::begin_unwind::h6365792021826883910
   5:     0x7f63e4e31abb - diagnostic::Handler::bug::h8da0df7c5604a97agwB
   6:     0x7f63e580467b - session::Session::bug::h415dd0bf24016cc06Dq
   7:     0x7f63e664c212 - check::FnCtxt<'a, 'tcx>::expr_ty::h0b20f2a038bc7a0c70o
   8:     0x7f63e6661427 - check::writeback::WritebackCx<'cx, 'tcx>.Visitor<'v>::visit_expr::h419b8916d070b338bdc
   9:     0x7f63e6661830 - check::writeback::WritebackCx<'cx, 'tcx>.Visitor<'v>::visit_expr::h419b8916d070b338bdc
  10:     0x7f63e66e99ea - check::check_bare_fn::h1ca89f422d965ecetwn
  11:     0x7f63e66e76f2 - check::CheckItemBodiesVisitor<'a, 'tcx>.Visitor<'tcx>::visit_item::h29ccc99d00d684a7wtn
  12:     0x7f63e67ab69a - check_crate::closure.38028
  13:     0x7f63e67a6a30 - check_crate::h22dcd95e17a2d96dXcC
  14:     0x7f63e7dd2cc8 - driver::phase_3_run_analysis_passes::h43926ceca86caa9fnGa
  15:     0x7f63e7db3dc5 - driver::compile_input::hb78754f2f33c01efQba
  16:     0x7f63e7e754d1 - run_compiler::h258d36d5501c1cdfz4b
  17:     0x7f63e7e73122 - boxed::F.FnBox<A>::call_box::h7239693171334256553
  18:     0x7f63e7e72659 - rt::unwind::try::try_fn::h14329119008520845439
  19:     0x7f63e7908ac8 - rust_try_inner
  20:     0x7f63e7908ab5 - rust_try
  21:     0x7f63e7e72908 - boxed::F.FnBox<A>::call_box::h17332056298259451807
  22:     0x7f63e7895041 - sys::thread::create::thread_start::h490278b5c3c0b49faqv
  23:     0x7f63e2125d14 - start_thread
  24:     0x7f63e74e453c - clone
  25:                0x0 - <unknown>
```

Version used
-----------------

```
rustc 1.0.0 (a59de37e9 2015-05-13) (built 2015-05-14)
binary: rustc
commit-hash: a59de37e99060162a2674e3ff45409ac73595c0e
commit-date: 2015-05-13
build-date: 2015-05-14
host: x86_64-unknown-linux-gnu
release: 1.0.0
```

**Comment by arielb1** [comment]
Fixed in 1.1.0.

**Comment by alexcrichton** [comment]
Indeed, thanks @arielb1!
